Documentation/ApiOverview/FlexForms/Index.rst

```diff
@@ -0,0 +1,380 @@
+                                        <type>check</type>
+                                    </config>
+                                </TCEforms>
+                            </settings.includeCategories>
+
+                            <!-- end of settings -->
+
+                        </el>
+                    </ROOT>
+                </sDEF>
+            </sheets>
+        </T3DataStructure>
+
+#. Add texts and translations
+
+   Add text for the language keys used in schema (LLL) to your language
+   files in :file:`Resources/Private/Languages`.
+
+   .. code-block:: xml
+
+       <trans-unit id="settings.registration.title">
+           <source>Settings for Registration plugin</source>
+       </trans-unit>
+       <trans-unit id="settings.registration.includeCategories">
+           <source>Include categories</source>
+       </trans-unit>
+
+
+#. The configuration schema is attached to one or more plugins
+
+   Here, the vendor name is **Myvendor, the extension key is **example**
+   and the plugin name is **Registration**.
+
+   in :file:`Configuration/TCA/Overrides/tt_content.php`:
+
+   .. code-block:: php
+
+      \TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin(
+          // The extension name (in UpperCamelCase) or the extension key (in lower_underscore)
+          'Myvendor.example',
+          // Plugin name: unique id in UpperCamelCase
+          'Registration',
+          // title
+          'LLL:EXT:example/Resources/Private/Language/Backend.xlf:registration_title'
+      );
+
+   * `API reference: registerPlugin <https://github.com/TYPO3/TYPO3.CMS/blob/ddedb065030de860de87e39636ed2664c7c531cd/typo3/sysext/extbase/Classes/Utility/ExtensionUtility.php#L104>`__
+
+   .. code-block:: php
+
+       $GLOBALS['TCA']['tt_content']['types']['list']['subtypes_addlist']['example_registration'] = 'pi_flexform';
+       \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPiFlexFormValue(
+           // plugin signature: <extension key> '_' <plugin name in lowercase>
+           'example_registration',
+           // Flexform configuration schema file
+           'FILE:EXT:example/Configuration/FlexForms/Registration.xml'
+       );
+
+   * `API reference: addPiFlexFormValue <https://github.com/TYPO3/TYPO3.CMS/blob/ddedb065030de860de87e39636ed2664c7c531cd/typo3/sysext/core/Classes/Utility/ExtensionManagementUtility.php#L1263>`__
+
+   .. tip::
+
+      The plugin signature is used in the database field `tt_content.ctype`
+      as well, when the tt_content record is saved. If you are confused about
+      how to handle underscores and upper / lowercase, check there to see
+      what your plugin signature is.
+
+      Also look on the page :ref:`extension-naming`.
+
+#. The settings can be read using :php:`$this->settings` in an
+   Extbase controller.
+
+   .. code-block:: php
+
+      $includeCategories = (bool) ($this->settings['includeCategories'] ?? false);
+
+
+Schema Examples
```

Comment:
I would not add to much examples here, as this duplicates information from TCA Reference and needs to be in sync.

Instead I would use https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/DataFormats/T3datastructure/Index.html, as mentioned above, and improve this place regarding structure.

Reply:
I think for newbies it is not so easy to map the writing in TCA (PHP) to the T3DataStructure data structure. That is why I added the examples. If you look at the examples in TCA and these ones side by side, I think it will be easier to understand how it works. I removed the input example though.